sentry/src/event_aggregator.rs

```diff
@@ -155,19 +155,32 @@ impl EventAggregator {
             update_targeting_rules(&app.pool, &channel_id, &new_rules).await?;
         }
 
-        events.iter().for_each(|ev| {
+        // Pre-computing all payouts once
+        let payouts: Vec<PayoutResult> = events
+            .iter()
+            .filter(|ev| ev.is_click_event() || ev.is_impression_event())
+            .map(|ev| get_payout(&app.logger, &record.channel, ev, &session))
+            .collect();
```

Comment:
Keep in mind, that we use `let payout = get_payout(logger, &channel, &ev, session)?;` in the `fn reduce()` which means that if one of the `get_payout` calls fails, then the whole `reduce()` fn. will fail (return an error). This should also be the case when we calculate the payouts here.

This is easily mitigated by: https://hoverbear.org/blog/a-useful-error-pattern/
If one of the `Result`s in the iterator is `Err` then the whole `collect` will return an `Err`. It will even error on the first `Err` that it encounters (called "short circuiting"), which means that it won't waste time to calculate all results first and then return an error.